src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,246 @@
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public function save(Key $key)
+    {
+        $resource = $this->getKeyResource($key);
+
+        $lockNodePath = self::LOCK_NODE.'/'.$resource; // For example: /PHP_LOCK_NODE/MESSAGE_1234
+        //This makes sure the node dies at the end of a session, thus releasing the lock to avoid starvation
+        $flags = Zookeeper::EPHEMERAL;
+
+        try {
+            $token = $this->getToken($key);
+            $this->createNewLock($lockNodePath, $token, array(), $flags);
+        } catch (LockConflictedException $ex) {
+            throw $ex;
+        } catch (Throwable $throwable) {
+            throw new LockAcquiringException($throwable);
+        }
+    }
+
+    /**
+     * @param \Symfony\Component\Lock\Key $key Key
+     */
+    public function delete(Key $key)
+    {
+        try {
+            $lockNodePath = self::LOCK_NODE.'/'.$this->getKeyResource($key);
+
+            if ($this->exists($key)) {
+                $this->zookeeper->delete($lockNodePath);
+            }
+        } catch (Throwable $throwable) {
+            // For Zookeeper Ephemeral Nodes, the node will be deleted upon session death. But, if we want to unlock
+            // the lock before proceeding further in the session, the client should be aware of this
+            throw new LockReleasingException($throwable);
+        }
+    }
+
+    /**
+     * Returns whether or not the resource exists in the storage.
+     *
+     * @param \Symfony\Component\Lock\Key $key Key
+     *
+     * @return bool
+     */
+    public function exists(Key $key): bool
+    {
+        $lockNodePath = self::LOCK_NODE.'/'.$this->getKeyResource($key);
+        if (!$this->zookeeper->exists($lockNodePath)) {
+            return false;
+        }
+
+        return $this->zookeeper->get($lockNodePath) === $this->getToken($key);
+    }
+
+    /**
+     * Waits until a key becomes free, then stores the resource.
+     *
+     * This store currently does not support this feature.
+     *
+     * @param \Symfony\Component\Lock\Key $key Key
+     *
+     * @throws NotSupportedException
+     */
+    public function waitAndSave(Key $key)
+    {
+        throw new NotSupportedException();
+    }
+
+    /**
+     * Extends the ttl of a resource.
+     *
+     * This store does not support this feature because, the locks are obtained using Ephemeral nodes
+     * that die after session death.
+     *
+     * @param \Symfony\Component\Lock\Key $key Key containing the resource to be locked
+     * @param float                       $ttl amount of second to keep the lock in the store
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+    }
+
+    /**
+     * @param int    $errNo  Error number
+     * @param string $errStr Error String
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public static function createNodeErrorHandler($errNo, $errStr)
```

Comment:
This doesn't create an error handler, it just always throws an exception. If you use it as static constructor for exceptions, it's better of in its own class and probably return the exception, rather than throw it. Return type can be void.

Reply:
The name might seem ambiguous but I mean it is a error handler for `createNode` function. Changed to `errorHandlerForCreateNode`. This is a very small function that I don't believe would require to be moved out into a separate class for simplicity reasons.